### ISSUE TITLE
Implement MDL aromaticity model option for handling SMIRFFs

### ISSUE THREAD

**Issue by davidlmobley** [open-forcefield-group/smarty#125]
In Christopher Bayly's parm@frosst SMIRFF draft, he's going to be relying on an alternate aromaticity model. Initially we thought he would need to kekulize all of the structures but it turns out that won't do what he needs, so instead we'll be using the MDL aromaticity model. 

It turns out that OEPrepareSearch clobbers whatever aromaticity model is used in the underlying OEMol, so we will need to add an option to SMIRFF to use this alternate aromaticity model and then bypass OEPrepareSearch by preparing our own search. Christopher has provided some example code for this.

``` python
mol = oechem.OEMol()
unique = True
for mol in ifs.GetOEMols():
    #print( mol.GetTitle(), mol.NumAtoms())
    oechem.OEClearAromaticFlags( mol)
    oechem.OEAssignAromaticFlags( mol, oechem.OEAroModel_MDL)
    # avoid running OEPrepareSearch or we lose MDL aromaticity, so instead:
    oechem.OEAssignHybridization( mol)
    oechem.OEAssignFormalCharges( mol)
    oechem.OEAssignImplicitHydrogens( mol)
    for triple in subschToATypes:
        subsch = triple[0]
        #oechem.OEPrepareSearch(mol, subsch)
        for match in subsch.Match(mol, unique):
            #print( 'have a match with', triple[2])
            #display( depictMatch(mol, match))
            for ma in match.GetAtoms():
                #print(ma.target.GetIdx(), triple[1], end=" ")
                ma.target.SetType( triple[1])
            #print('')
    oechem.OEWriteConstMolecule( ofs, mol)
    for atom in mol.GetAtoms():
        ofsTypes.write( '%s %s\n' % (mol.GetTitle(), atom.GetType()) )
```

This would be in lieu of the `kekulize` option proposed in https://github.com/open-forcefield-group/smarty/issues/53; probably we'd trigger via `aromaticitymodel=MDL` or something along those lines. Will come up with a concrete proposal when i get ready to implement (which will need to be soon).

**Comment by davidlmobley** [comment]
Ah, actually, it looks like we are not currently using `OEPrepareSearch` in `ForceField`; the search is done via whatever aromaticity model/preparation is applied to the provided OEMol. This means an interim solution for @cbayly13 is to apply the desired aromaticity model prior to providing the oemol(s) to `ForceField`.

However, I should still implement something to invoke this. In #53 I'd proposed:

``` XML
<SMIRFF kekulize="True">
...
</SMIRFF>
```

Here, I think this means we should do the same but use

``` XML
<SMIRFF aromaticity_model="OEAroModel_MDL">
...
</SMIRFF>
```

and this should trigger the behavior above to be applied to the provided oemols. 

The easiest/most sensible place to invoke this would be, it seems, by extending `getSMIRKSMatches_OEMol` to take an optional `aromaticity_model` argument which would apply the above code, then to allow `ForceField` to store the desired aromaticity model if provided in the XML.

**Comment by jchodera** [comment]
Sounds great!

Make sure to make a deep copy of the OEMols first so we don't change the
ones the user passes in.

**Comment by davidlmobley** [comment]
Resolved by #127
